test/collector.js

```diff
@@ -209,3 +210,105 @@ metatests.test('Collector: error in then chain', (test) => {
     },
   );
 });
+
+metatests.test('Collector: collect with schema validation', async (test) => {
+  const schema = Schema.from({
+    firstname: 'string',
+    lastname: 'string',
+  });
+  const ac1 = collect(['firstname', 'lastname'], { schema });
+
+  ac1.set('firstname', undefined);
+  ac1.set('lastname', 1);
+
+  const expectedResult1 = new Error(
+    'Invalid keys type: Field "firstname" not of expected type: string; Field "lastname" not of expected type: string',
+  );
+
+  try {
+    await ac1;
```

Comment:
In all tests where you should catch an error, you should also throw an extra error if that test passed without the expected error. Otherwise, you simply won't notice that the test went wrong(after `npm run test` you will see simply one test less, that is, all test will go successfully without errors). It was recently added in all our tests.

So after all `await` in `try` you should add this line:
`test.error(new Error('Should not be executed'));`